### DISCUSSION TITLE
Vanchor, with no motor controll

### DISCUSSION THREAD

**Discussion by dennismilton** [AlexAsplund/Vanchor#23] [Ideas]
Hi,

Lets assume, there is already installed a autopilot, and steppermotor/hydralic controll etc.. (f.ex. mine is garmin reactor 40 with ghc20 head)

And my plotter is iOS app Navionics - wifi onboard and my navionics receives nmea data correctly - but it cannot control the autopilot as it doesnt send xte, apb etc….

Heres the question :) - would it be possible to import a gpx route, activate it - and let your app send out the autopilot commands to my nmea network? - even if its has no motor controller ? (i know my ghc20 can use data from nmea network)

/Dennis

**Comment by AlexAsplund** [comment]
That should theoretically be possible yes or just borrowing the stuff that does the conversion :)
But i think that the iNavX app can send APB as well if that's an alternative.

The main part is here:
https://github.com/AlexAsplund/Vanchor/blob/96203badf381f7afdd1f71c191e0f1718c44046b/src/vanchor/functions/autopilot.py
``` python
        dest = route["wp{}".format(index)][1]
        curr_to_dest = self.main.tools.converter.get_bearing(curr, dest)
        start_to_dest = self.main.tools.converter.get_bearing(start, dest)

        if curr_to_dest["Distance"] <= arrival_radius:
            self.logger.info(
                "Arrived at waypoint '{}'. Selecting next waypoint (if it exists)".format(
                    waypoint_name
                )
            )
            self.main.emitter.emit("autopilot.waypoint.arrived", waypoint_name)
            return

        self.logger.debug(
            "Calculating distance and bearing. Start:{} Dest:{} Curr:{}".format(
                start, dest, curr
            )
        )
        xte = self.main.tools.geo.getCrossTrackDistance(start, dest, curr)

        self.logger.debug("StartToDest: {}".format(start_to_dest))
        self.logger.debug("CurrToDest: {}".format(curr_to_dest))
        if xte < 0:
            direction = "R"
        else:
            direction = "L"

        self.logger.debug("XTE is: {}".format(xte))

        apb_data = (
            "A",
            "A",
            str(round(abs(xte), 3)),
            direction,
            "M",
            "T",
            "T",
            str(start_to_dest["ForwardAzimuth"] % 360),
            "M",
            str(waypoint_name),
            str(curr_to_dest["ForwardAzimuth"] % 360),
            "M",
            direction,
            "M",
        )
```
